source/server/listener_manager_impl.cc

```diff
@@ -1022,7 +1022,9 @@ Network::DrainableFilterChainSharedPtr ListenerFilterChainFactoryBuilder::buildF
                                      transport_socket.DebugString()));
   }
 #else
-  UNREFERENCED_PARAMETER(listener_);
+  // When QUIC is compiled out it should not be possible to configure either the QUIC transport
+  // socket or the QUIC listener and get to this point.
```

Comment:
mind filing a tracking issue for upstream/downstream quic to have tests in the compile time options build that we fail gracefully if H3 is configured upstream/downstream while compiled out?

Reply:
I made a note here: https://github.com/envoyproxy/envoy/issues/12829#issuecomment-809538430

I will work on this in the follow up PR I have ongoing to remove codec/client extension points for QUIC.